### ISSUE TITLE
New logs with keeps-going are hard to parse

### ISSUE THREAD

**Issue by malfet** [pytorch/test-infra#4836]
For example, have a look at https://github.com/pytorch/pytorch/actions/runs/7250984617/job/19752690096

First of all, it feels like the test, once it failed re-run 9 rather than 3 times, otherwise I can't explain why `RERUN` message for the same test appears 6 times in the log:
```
% grep "test_output_match_transpose_cpu_complex64 ('RERUN'," log.log
2023-12-18T16:52:47.7888700Z test_mps.py::TestConsistencyCPU::test_output_match_transpose_cpu_complex64 ('RERUN', {'yellow': True}) [0.0064s] [ 11%]
2023-12-18T16:52:47.7891020Z test_mps.py::TestConsistencyCPU::test_output_match_transpose_cpu_complex64 ('RERUN', {'yellow': True}) [0.0059s] [ 11%]
2023-12-18T16:52:47.8017980Z test_mps.py::TestConsistencyCPU::test_output_match_transpose_cpu_complex64 ('RERUN', {'yellow': True}) [0.3396s] [  0%]
2023-12-18T16:52:47.8088250Z test_mps.py::TestConsistencyCPU::test_output_match_transpose_cpu_complex64 ('RERUN', {'yellow': True}) [0.0070s] [  0%]
2023-12-18T16:52:47.8263940Z test_mps.py::TestConsistencyCPU::test_output_match_transpose_cpu_complex64 ('RERUN', {'yellow': True}) [0.3397s] [  0%]
2023-12-18T16:52:47.8264780Z test_mps.py::TestConsistencyCPU::test_output_match_transpose_cpu_complex64 ('RERUN', {'yellow': True}) [0.0068s] [  0%]
```

Also message about items run in the shard appears over and over and over again:
```
% grep "items in this shard:" log.log|wc -l
      44
```

**Comment by clee2000** [comment]
Tests rerun 9 times, 3 times within a process x 3 process retries.  

The message for "items in this shard" lists the tests that pytest wants to run, it's nice for when your test process stops early due to failure but you want to check that your test exists.  I'll try to reduce the number of times it gets printed though, since most of it is repeated.

**Comment by clee2000** [comment]
Frequency of items in shard line has been reduced in https://github.com/pytorch/pytorch/pull/117069
Closing since the 9 reruns is intentional
